### PR TITLE
doc: lib: nrf_modem: updates

### PR DESCRIPTION
**Pull request by lemrey**
Small updates.

### CODE REVIEW

File: doc/nrf/libraries/modem/nrf_modem_lib.rst [hunk 1]
```diff
@@ -250,26 +248,24 @@ Other configurations set in the devicetree, such as the current speed, are overw
 
 Modem fault handling
 ********************
-If a fault occurs in the modem, the application is notified through the fault handler function that is registered with the Modem library during initialization.
-This enables the application to read the fault reason (in some cases the modem's program counter) and take appropriate action.
+When a fault occurs in the modem, the application is notified through the fault handler function that is registered with the Modem library during initialization.
```
Comment (eivindj-nordic) [inline doc/nrf/libraries/modem/nrf_modem_lib.rst:251]: `When` makes it sound like this is something we expect to happen.
```suggestion
If a fault occurs in the modem, the application is notified through the fault handler function that is registered with the Modem library during initialization.
```

**Comment by nordicjm** [comment]
@lemrey needs rebase and comments addressed

File: doc/nrf/libraries/modem/nrf_modem_lib.rst (hunk 1 above)
Comment (lemrey) [reply doc/nrf/libraries/modem/nrf_modem_lib.rst]: Ok, updated

File: doc/nrf/libraries/modem/nrf_modem_lib.rst [hunk 2]
```diff
@@ -250,26 +248,24 @@ Other configurations set in the devicetree, such as the current speed, are overw
...
+* :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_DO_NOTHING` - Let the fault handler log the Modem fault and return (default).
+* :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_RESET_MODEM` - Let the fault handler schedules a workqueue task to reinitialize the modem and Modem library.
+* :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_APPLICATION_SPECIFIC` - Let the fault handler function :c:func:`nrf_modem_fault_handler` be defined by the application, outside of the Modem library integration layer.
 
 Implementing a custom fault handler
 ===================================
 
-If you want to implement a custom fault handler, consider the following points:
+When implementing a custom fault handler, enable the :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_APPLICATION_SPECIFIC` Kconfig option
+and provide an implementation of the :c:func:`nrf_modem_fault_handler` function, considering the following points:
```
[the remark this answers is not in the document]
Comment (lemrey) [reply doc/nrf/libraries/modem/nrf_modem_lib.rst]: Thanks,  updated

File: doc/nrf/libraries/modem/nrf_modem_lib.rst [hunk 3]
```diff
@@ -250,26 +248,24 @@ Other configurations set in the devicetree, such as the current speed, are overw
...
-This enables the application to read the fault reason (in some cases the modem's program counter) and take appropriate action.
+When a fault occurs in the modem, the application is notified through the fault handler function that is registered with the Modem library during initialization.
+This lets the application read the fault reason (in some cases the modem's program counter) and take the appropriate action.
 
-On initialization, the Modem library integration layer registers the :c:func:`nrf_modem_fault_handler` function through the Modem library initialization parameters.
-The behavior of the :c:func:`nrf_modem_fault_handler` function is controlled with the :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT` Kconfig option.
-The Modem library integration layer provides the following three options for :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT` Kconfig option:
+On initialization (during :c:func:`nrf_modem_lib_init`), the Modem library integration layer registers the :c:func:`nrf_modem_fault_handler` function through the Modem library initialization parameters.
```
[the remark this answers is not in the document]
Comment (lemrey) [reply doc/nrf/libraries/modem/nrf_modem_lib.rst]: Ok

**Review by lemrey — COMMENTED** [review]
@divipillai please have another look

File: doc/nrf/libraries/modem/nrf_modem_lib.rst [hunk 4]
```diff
@@ -251,25 +249,22 @@ Other configurations set in the devicetree, such as the current speed, are overw
...
+The behavior of the :c:func:`nrf_modem_fault_handler` function is controlled with the three following KConfig options:
 
-* :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_DO_NOTHING` - This is the default Kconfig option that lets the fault handler log the Modem fault and return.
-* :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_RESET_MODEM`- This Kconfig option schedules a workqueue task to reinitialize the modem and Modem library.
-* :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_APPLICATION_SPECIFIC`- This Kconfig option results in a call to the :c:func:`nrf_modem_fault_handler` function that is defined in the application, outside of the Modem library integration layer.
+* :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_DO_NOTHING` - Let the fault handler log the Modem fault and return (default).
+* :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_RESET_MODEM` - Let the fault handler schedules a workqueue task to reinitialize the modem and Modem library.
+* :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_APPLICATION_SPECIFIC` - Let the fault handler function :c:func:`nrf_modem_fault_handler` be defined by the application, outside of the Modem library integration layer.
```
Comment (divipillai) [inline doc/nrf/libraries/modem/nrf_modem_lib.rst:259]:
```suggestion
* :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_DO_NOTHING` - This option lets the fault handler log the Modem fault and return (default).
* :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_RESET_MODEM` - This option lets the fault handler schedule a workqueue task to reinitialize the modem and Modem library.
* :kconfig:option:`CONFIG_NRF_MODEM_LIB_ON_FAULT_APPLICATION_SPECIFIC` - This option lets the fault handler function :c:func:`nrf_modem_fault_handler` be defined by the application, outside of the Modem library integration layer.
```

**Review by divipillai — APPROVED** [review]
A minor suggestion. Rest all LGTM.

File: doc/nrf/libraries/modem/nrf_modem_lib.rst (hunk 4 above)
Comment (lemrey) [reply doc/nrf/libraries/modem/nrf_modem_lib.rst]: Thanks, updated.

**Review by eivindj-nordic — APPROVED** [review]

**Comment by lemrey** [comment]
SHA updated

**Pull request merged**